docs/concepts.rst

```diff
@@ -0,0 +1,112 @@
+.. _concepts:
+
+Concepts
+========
+
+Reusable analyses
+-----------------
+
+The revalidation, reinterpretation and reuse of research data analyses requires
+having access not only to the original experimental datasets and the analysis
+software, but also to the operating system environment and the computational
+workflow steps which were used by the researcher to produce the original
+scientific results in the first place.
+
+Four questions
+--------------
+
+REANA helps to make the research analysis reusable by providing a structure
+helping to answer the "Four Questions":
+
+1. What is your input data?
+
+   - input files
+   - input parameters
+   - live database calls
+
+2. What is your environment?
+
+   - operating systems
+   - software packages and libraries
+   - CPU and memory resources
+
+3. Which code analyses it?
+
+   - analysis frameworks
+   - custom analysis code
+   - Jupyter notebooks
+
+4. Which steps did you take?
+
+   - simple linear commands
+   - complex workflows
+   - local or remote workflow step execution
+
+Analysis structure
+-------------------
+
+It is advised to structure your analysis code using the "inputs", "code",
+"outputs", "environments", "workflows" strategy. For example:
+
+.. code-block:: console
+
+   $ ls -l .
+   inputs/mydata.csv
+   code/mycode.py
+   environments/myenvironment/Dockerfile
```

Comment:
I don't know if it is only me, but this "example" sounds kind of ok in general, but `myenvironment` specifically sounds way too abstract

Reply:
Yeah, it looks a bit cryptic... What about using `mypython` or `mybuntu` or something more understandable like that?